tests/context_tests.rs

```diff
@@ -0,0 +1,1147 @@
+        let mut context = Context::new(Tcti::Mssim).unwrap();
+        context
+            .start_auth_session(
+                NO_SESSIONS,
+                ESYS_TR_NONE,
+                ESYS_TR_NONE,
+                &[],
+                TPM2_SE_HMAC,
+                utils::TpmtSymDefBuilder::aes_256_cfb(),
+                TPM2_ALG_SHA256,
+            )
+            .unwrap();
+    }
+
+    #[test]
+    fn test_nonce_sess() {
+        let mut context = Context::new(Tcti::Mssim).unwrap();
+        context
+            .start_auth_session(
+                NO_SESSIONS,
+                ESYS_TR_NONE,
+                ESYS_TR_NONE,
+                &[
+                    128, 85, 22, 124, 85, 9, 12, 55, 23, 73, 1, 244, 102, 44, 95, 39, 10,
+                ],
+                TPM2_SE_HMAC,
+                utils::TpmtSymDefBuilder::aes_256_cfb(),
+                TPM2_ALG_SHA256,
+            )
+            .unwrap();
+    }
+
+    #[test]
+    fn test_long_nonce_sess() {
+        let mut context = Context::new(Tcti::Mssim).unwrap();
+        context
+            .start_auth_session(
+                NO_SESSIONS,
+                ESYS_TR_NONE,
+                ESYS_TR_NONE,
+                &[
+                    231, 97, 201, 180, 0, 1, 185, 150, 85, 90, 174, 188, 105, 133, 188, 3, 206, 5,
+                    222, 71, 185, 1, 209, 243, 36, 130, 250, 116, 17, 0, 24, 4, 25, 225, 250, 198,
+                    245, 210, 140, 23, 139, 169, 15, 193, 4, 145, 52, 138, 149, 155, 238, 36, 74,
+                    152, 179, 108, 200, 248, 250, 100, 115, 214, 166, 165, 1, 27, 51, 11, 11, 244,
+                    218, 157, 3, 174, 171, 142, 45, 8, 9, 36, 202, 171, 165, 43, 208, 186, 232, 15,
+                    241, 95, 81, 174, 189, 30, 213, 47, 86, 115, 239, 49, 214, 235, 151, 9, 189,
+                    174, 144, 238, 200, 201, 241, 157, 43, 37, 6, 96, 94, 152, 159, 205, 54, 9,
+                    181, 14, 35, 246, 49, 150, 163, 118, 242, 59, 54, 42, 221, 215, 248, 23, 18,
+                    223,
+                ],
+                TPM2_SE_HMAC,
+                utils::TpmtSymDefBuilder::aes_256_cfb(),
+                TPM2_ALG_SHA256,
+            )
+            .unwrap_err();
+    }
+
+    #[test]
+    fn test_bound_sess() {
+        let mut context = Context::new(Tcti::Mssim).unwrap();
+        let prim_key_handle = context
+            .create_primary_key(
+                ESYS_TR_RH_OWNER,
+                &utils::get_rsa_public(true, true, false, 2048),
+                &[],
+                &[],
+                &[],
+                &[],
+            )
+            .unwrap();
+
+        context
+            .start_auth_session(
+                NO_SESSIONS,
+                prim_key_handle,
+                prim_key_handle,
+                &[],
+                TPM2_SE_HMAC,
+                utils::TpmtSymDefBuilder::aes_256_cfb(),
+                TPM2_ALG_SHA256,
+            )
+            .unwrap();
+    }
+
+    #[ignore]
```

Comment:
Maybe it would be nice to explain with a comment why this test is ignored?

Reply:
Done!

Reply:
Super